### PR TITLE
Migrate CI to GitHub Actions

### PR DESCRIPTION
**Pull request by r6eve**
Travis CI is sometimes slow to start and changes the provided JDK versions drastically https://github.com/chrovis/cljam/pull/170. To better achieve stability, it's time to switch CI from Travis CI to GitHub Actions.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        key: ${{ runner.os }}-m2-${{ hashFiles('**/project.clj') }}
+        restore-keys: |
+          ${{ runner.os }}-m2-
+    - name: Install Dependencies
+      run: lein deps
+    - name: Run tests
+      run: |
+        lein with-profile +dev:+1.8:+1.9 test
+        lein with-profile dev bin
+        target/cljam version
```
Comment (r6eve) [inline .github/workflows/main.yml]: Duplicated code here. I tried to include EA in Java's metrix, and found it's impossible for now.

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    runs-on: ubuntu-18.04
+    steps:
+    - uses: actions/checkout@v1
+    - name: Setup Java
+      uses: actions/setup-java@v1
+      with:
+        java-version: '1.8'
+        architecture: x64
+    - name: Deploy
+      run: ./deploy-snapshot.sh
```
Comment (alumi) [inline .github/workflows/deploy.yml]: I want this workflow to be executed after all jobs in the `main` workflow have been finished successfully.
I think keeping workflows separated has advantages of maintainability and security but it seems we can't define dependencies for workflows for now.
Do we have to merge this job into `main` with [`if`](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idif) directive?

Comment (r6eve) [reply .github/workflows/deploy.yml]: Yes. As far as I know, there's no way except `if` directive.
I incorporated the `deploy` job into the `main` workflow with some constraints. When pushing master, if any of `main` tests fails, the deploy is not executed.

**Comment by r6eve** [comment]
Thanks for reviewing.

**Pull request merged**